app/controllers/admin/merchants_controller.rb

```diff
@@ -2,5 +2,6 @@ class Admin::MerchantsController < Admin::BaseController
   def show
     @merchant = User.find(params[:id])
     @orders = Order.pending_orders(@merchant)
+    redirect_to admin_user_path(@merchant) if @merchant.registered?
```

Comment:
I think because my last branch hasn't been merged yet there are a bunch of changes that appear. It's actually just this one line plus the test for it.

Reply:
Seems like a quick fix. Good work!